### PR TITLE
apache-couchdb 2.3.1

### PR DESCRIPTION
**Pull request by big-r81**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Creating a new formula for Apache CouchDB 2.3.1. to have an updated version to Formula/couchdb.rb (1.7.2)

### CODE REVIEW

**Comment by fxcoudert** [comment]
Please make this an update to the existing formula.

**Comment by big-r81** [comment]
hi,

i have updated the formula.
For discussion:
I think it is better to create a new formula for the 2.3 branch of couchdb to have both versions available in homebrew. then you don't have to deal with upgrade procedures of the db when upgrading the formula...

**Comment by SMillerDev** [comment]
Will upstream actively support both versions of couchdb?

**Comment by fxcoudert** [comment]
Closing due to lack of activity

**Comment by big-r81** [comment]
The lack of activity is not from my side. There are two things open:

- For discussion:
I think it is better to create a new formula for the 2.3 branch of couchdb to have both versions available in homebrew. then you don't have to deal with upgrade procedures of the db when upgrading the formula...
- Will upstream actively support both versions of couchdb?

But there are no naswers from the homebrew team...

**Comment by fxcoudert** [comment]
> Will upstream actively support both versions of couchdb?

@SMillerDev is a Homebrew maintainer, and he asked that question to be able to decide best whether we want two versions or not.

File: Formula/couchdb.rb [hunk 1]
```diff
@@ -1,71 +1,59 @@
 class Couchdb < Formula
-  desc "Document database server"
+  desc "Apache CouchDB database server"
   homepage "https://couchdb.apache.org/"
-  url "https://www.apache.org/dyn/closer.cgi?path=/couchdb/source/1.7.2/apache-couchdb-1.7.2.tar.gz"
-  mirror "https://archive.apache.org/dist/couchdb/source/1.7.2/apache-couchdb-1.7.2.tar.gz"
-  sha256 "7b7c0db046ded544a587a8935d495610dd10f01a9cae3cd42cf88c5ae40bc431"
-  revision 1
-
-  bottle do
```
Comment (fxcoudert) [inline Formula/couchdb.rb]: leave the bottle block untouched

File: Formula/couchdb.rb [hunk 2]
```diff
@@ -1,71 +1,59 @@
...
-  end
-
-  depends_on "erlang@19"
+  url "https://www.apache.org/dyn/closer.lua?path=/couchdb/source/2.3.1/apache-couchdb-2.3.1.tar.gz"
+  sha256 "43eb8cec41eb52871bf22d35f3e2c2ce5b806ebdbce3594cf6b0438f2534227d"
+
+  depends_on "autoconf" => :build
+  depends_on "autoconf-archive" => :build
+  depends_on "automake" => :build
+  depends_on "erlang@21" => :build
```
Comment (fxcoudert) [inline Formula/couchdb.rb]: Does it not build with more recent erlang?

File: Formula/couchdb.rb [hunk 3]
```diff
@@ -1,71 +1,59 @@
...
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
   depends_on "icu4c"
+  depends_on "openssl"
```
Comment (fxcoudert) [inline Formula/couchdb.rb]: Use OpenSSL 1.1 (`openssl@1.1`)

Comment (big-r81) [reply Formula/couchdb.rb]: updated

File: Formula/couchdb.rb (hunk 2 above)
Comment (big-r81) [reply Formula/couchdb.rb]: no, support for erlang 22+ only for master

File: Formula/couchdb.rb (hunk 1 above)
Comment (big-r81) [reply Formula/couchdb.rb]: get back bottle block

**Comment by big-r81** [comment]
- Version 1.7.x ist deprecated long ago (I asked the developers).

**Comment by SMillerDev** [comment]
In that case we don't want to have two formula because homebrew doesn't ship versions that are unsupported by upstream. If there is a way to upgrade from one to the other maybe the Twitter for homebrew can tweet a warning when we merge this to warm people at least.

**Comment by big-r81** [comment]
i don't know why the jenkins checks are failing...

**Comment by fxcoudert** [comment]
> i don't know why the jenkins checks are failing...

Please rebase as a single commit on top of master

**Comment by big-r81** [comment]
superseeded by #46405

**Pull request closed without merging**